### PR TITLE
add resource group memory statistics

### PR DESCRIPTION
**Pull request by SmartKeyerror**
Add memory_usage into the function `pg_resgroup_get_status()` result.

```
postgres=# select * from gp_toolkit.gp_resgroup_status_per_host;
    rsgname    | groupid | hostname | cpu_usage | memory_usage 
---------------+---------+----------+-----------+--------------
 admin_group   |    6438 | zero     |      0.09 |         5.55
 default_group |    6437 | zero     |      0.00 |         0.00
 system_group  |    6441 | zero     |      0.01 |        40.55
(3 rows)
``` 

It'll return the memory_usage in megabytes, to help the customer address memory problems.

### CODE REVIEW

**Comment by kainwen** [comment]
Thanks for the work! This PR modifies catalog, lets bump catversion.

File: src/backend/utils/resgroup/resgroup_helper.c [hunk 1]
```diff
@@ -80,19 +81,21 @@ calcCpuUsage(StringInfoData *str,
 static void
 getResUsage(ResGroupStatCtx *ctx, Oid inGroupId)
 {
-	int64 *usages;
```
Comment (kainwen) [inline src/backend/utils/resgroup/resgroup_helper.c:83]: better to change a new line to def a new var.

File: src/include/utils/cgroup.h [hunk 2]
```diff
@@ -248,6 +252,7 @@ typedef struct CGroupOpsRoutine
 	setcpupriority_function	setcpupriority;
 
 	getcpuusage_function 	getcpuusage;
+	getmemoryusage_memory	getmemoryusage;
```
Comment (yanwr1) [inline src/include/utils/cgroup.h:255]: maybe getmemoryusage_function?

Comment (SmartKeyerror) [reply src/include/utils/cgroup.h:255]: Nice catch!

**Review by yanwr1 — APPROVED** [review]

**Pull request merged**